pinot-spi/src/main/java/org/apache/pinot/spi/stream/RowMetadata.java

```diff
@@ -31,6 +32,8 @@
 @InterfaceAudience.Public
 @InterfaceStability.Evolving
 public interface RowMetadata {
+  GenericRow EMPTY_ROW = new GenericRow();
```

Comment:
I agree with you about the confusion. I was trying not to blow up/break existing interfaces in pinot-spi. 

Let me create a separate PR that just renames `RowMetadata` into `StreamMessageMetadata`. I want to avoid using the term "row" when we talk about input from a stream. I find that confusing :/

Reply:
I tried to remove `RowMetadata`. But it looks like it is being passed around in other interfaces like `MutableSegment` .  

I think it is fine to leave the interface as it will be convenient to include in other interface APIs and also, possibly extend it. I am wondering if there is a lot of value in distinguishing `RowMetadata` from a `StreamMessage`'s metadata. 

`RowMetadata` can refer to Pinot related row metadata (Ex. time at which a record was indexed in Pinot) . Where as, `StreamMessageMetadata` continues to simply hold the incoming record related metadata (ex. when record was ingested into upstream etc).  Wdyt @npawar ?

Reply:
and StreamMessageMetadata would still extend RowMetadata or you'd completely separate and unlink the 2? 
If it's hard to remove, i'm okay with leaving this as is for now

Reply:
+1, we should find ways of removing rowmetadata with the introduction of more generic class. Can we mark it deprecated?

Reply:
I will follow up with this in a separate PR (deprecation/deletion of `RowMetadata`)

Reply:
Created an issues for this https://github.com/apache/pinot/issues/9470 . Let's resolve this conversation and follow-up.